clients/python/tensorzero/types.py

```diff
@@ -468,3 +468,61 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
             stacklevel=2,
         )
         super().__init__(*args, **kwargs)
+
+
+@dataclass
+class Tool:
+    description: str
+    parameters: Any
+    name: str
+    strict: bool
+
+
+@dataclass
+class ToolParams:
+    tools_available: List[Tool]
+    tool_choice: str
+    parallel_tool_calls: Optional[bool]
+
+
+@dataclass
+class ChatInferenceDatapoint:
+    dataset_name: str
+    function_name: str
+    id: UUID
+    episode_id: Optional[UUID]
+    input: InferenceInput  # Should be ResolvedInput
+    output: Optional[List[ContentBlock]]
+    tool_params: Optional[ToolParams] = None
+    tags: Optional[Dict[str, str]] = None
+    auxiliary: str = ""
+    source_inference_id: Optional[UUID] = None
+    staled_at: Optional[str] = None
+    is_deleted: bool = False
+
+
+@dataclass
+class JsonInferenceDatapoint:
+    dataset_name: str
+    function_name: str
+    id: UUID
+    episode_id: Optional[UUID]
+    input: InferenceInput
+    output: Optional[JsonInferenceOutput]
+    output_schema: Optional[Any] = None
+    tags: Optional[Dict[str, str]] = None
+    auxiliary: str = ""
+    source_inference_id: Optional[UUID] = None
+    staled_at: Optional[str] = None
+    is_deleted: bool = False
+
+
+Datapoint = Union[ChatInferenceDatapoint, JsonInferenceDatapoint]
```

Comment:
Should we call these `ChatDatapoint` and `JsonDatapoint`? Datapoints won't necessarily come from an inference.

Reply:
the tables are called ChatInferenceDatapoint and JsonInferenceDatapoint and I think that's because even if they aren't _from_ inferences they are _for_ inferences. I don't feel super strongly except it's easier to deal with a smaller number of names.